include/treelite/c_api_runtime.h

```diff
@@ -96,75 +41,37 @@ TREELITE_DLL int TreeliteBatchGetDimension(void* handle,
  * \param out handle to predictor
  * \return 0 for success, -1 for failure
  */
-TREELITE_DLL int TreelitePredictorLoad(const char* library_path,
-                                       int num_worker_thread,
-                                       PredictorHandle* out);
+TREELITE_DLL int TreelitePredictorLoad(
+    const char* library_path, int num_worker_thread, PredictorHandle* out);
 /*!
  * \brief Make predictions on a batch of data rows (synchronously). This
  *        function internally divides the workload among all worker threads.
  * \param handle predictor
- * \param batch a batch of rows (must be of type SparseBatch or DenseBatch)
- * \param batch_sparse whether batch is sparse (1) or dense (0)
+ * \param batch the data matrix containing a batch of rows
  * \param verbose whether to produce extra messages
  * \param pred_margin whether to produce raw margin scores instead of
  *                    transformed probabilities
- * \param out_result resulting output vector; use
- *                   TreelitePredictorQueryResultSize() to allocate sufficient
- *                   space
+ * \param out_result Resulting output vector. This pointer must point to an array of length
+ *                   TreelitePredictorQueryResultSize() and of type
+ *                   TreelitePredictorQueryLeafOutputType().
```

Comment:
I feel it a bit confusing what values to pass to "out_result" (compared to the previous API, which was just a float).
My idea is to add a function in the API that allocates an area for out_result.
e.g. int TreelitePredictorOutput(PredictorHandle handle, PredictorOutputHandle* out_result)

Reply:
Good suggestion. I'll add it.

Reply:
I added C API functions `TreeliteCreatePredictorOutputVector()` and `TreeliteDeletePredictorOutputVector()`. I also added detailed instructions on accessing element values from the output vector.